google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java

```diff
@@ -204,10 +210,13 @@ public static ClientContext createClientContext(EnhancedBigtableStubSettings set
             ? ((InstantiatingGrpcChannelProvider) builder.getTransportChannelProvider()).toBuilder()
             : null;
 
-    if (builder.getEnableRoutingCookie() && transportProvider != null) {
-      // TODO: this also need to be added to BigtableClientFactory
-      // patch cookies interceptor
-      transportProvider.setInterceptorProvider(() -> ImmutableList.of(new CookiesInterceptor()));
+    Set<ConnectionErrorCountInterceptor> connectionErrorCountInterceptors =
+        Collections.newSetFromMap(new WeakHashMap<>());
+    setupConnectionErrorCountTask(builder, connectionErrorCountInterceptors);
+
+    if (transportProvider != null) {
+      transportProvider.setInterceptorProvider(
+          () -> getInterceptors(builder, connectionErrorCountInterceptors));
```

Comment:
Another bug here (carried over from my implementation :(). If someone already set an interceptor on the transport provider, they will lose their settings. We should get the interceptors the user set and merge it.

Reply:
discussed offline, the interceptor is a private field in InstantiatingGrpcChannelProvider without a getter. Its getInterceptors() method only gets called inside createSingleChannel(), so seems like we can't access the original one if it's been set. Leaving this as is for now.

Reply:
I think you can work around this by using the ChannelCOnfigurator instead of InterceptorProvider and set the interceptor directly on the ManagedChannelBuilder